desc/equilibrium/coords.py

```diff
@@ -115,33 +124,27 @@ def map_coordinates(  # noqa: C901
     # TODO (#1382): make this work for permutations of in/out basis
     if outbasis == ("rho", "theta", "zeta"):
         if inbasis == ("rho", "alpha", "zeta"):
+            errorif(
+                np.isfinite(period[1]),
+                msg=f"Period must be ∞ for inbasis={inbasis}, but got {period[1]}.",
+            )
             if "iota" in kwargs:
                 iota = kwargs.pop("iota")
+            elif "profiles" in kwargs:
+                iota = eq._compute_iota_under_jit(coords, params, **kwargs)
             else:
-                if profiles["iota"] is None:
-                    profiles["iota"] = eq.get_profile(
-                        ["iota", "iota_r"], params=params, **kwargs
-                    )
-                iota = profiles["iota"].compute(Grid(coords, sort=False, jitable=True))
-            return _map_clebsch_coordinates(
-                coords=coords,
-                iota=iota,
-                L_lmn=params["L_lmn"],
-                L_basis=eq.L_basis,
-                guess=guess[:, 1] if guess is not None else None,
-                period=period[1] if period is not None else np.inf,
-                tol=tol,
-                maxiter=maxiter,
-                full_output=full_output,
-                **kwargs,
-            )
+                iota = eq._compute_iota_under_jit(coords, params, profiles, **kwargs)
+            rho, alpha, zeta = coords.T
+            omega = 0  # TODO(#568)
+            coords = jnp.column_stack([rho, alpha + iota * (zeta + omega), zeta])
```

Comment:
If zeta is the generalized toroidal angle, don't we assume zeta = phi + Omega where phi is the cylindrical toroidal angle?
So shouldn't theta_PEST = alpha + iota * (zeta - omega)?